lib/session.js

```diff
@@ -378,7 +378,7 @@ function longSms(pduObj) {
 	                                                                              // 4th octet and length is
 	                                                                              // header size -2 octets
 	    partsCount    = pduObj.params.short_message[2 + csmsReference.length + 1],
-	    partNr        = pduObj.params.short_message[2 + csmsReference.length + 1],
+	    partNr        = pduObj.params.short_message[2 + csmsReference.length + 2],
```

Comment:
I have generalized it so it would work for both. So in case scenario from the referred document 
would be like following.
- First "2" is skipping the first 3 octets which remain constant
- csmsReference.length is the UDH size which we skip
- then we need to skip total parts count field which is 1 octet and then comes the actual partNr